go/helper-image/Dockerfile

```diff
@@ -19,8 +19,6 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 RUN curl --location --output delve.tar.gz https://github.com/go-delve/delve/archive/v$DELVE_VERSION.tar.gz \
   && tar xzf delve.tar.gz \
   && mv delve-$DELVE_VERSION delve-source
-COPY delve-*.patch .
-RUN patch -p0 -d delve-source < delve-as-options.patch
```

Comment:
I'm curious why you decided to just delete this rather than update it to apply to Delve 1.23.0. Are the issues Brian mentioned in [this PR](https://github.com/go-delve/delve/pull/2816) no longer relevant? From reading that PR, it sounds to me like this will make the image:

* [not work with binaries compiled with Go >= 1.24](https://github.com/go-delve/delve/blob/56e9b85f40ccd5ea92186ac54dba79a2f55d2197/pkg/goversion/compat.go#L13), even though it may turn out to be fully compatible. (I'm guessing this is why you had to delete all the tests for the older versions?)
* not work at all because the connecting user is different. (Perhaps the tests in this repo don't really cover this issue? Or maybe there's some other reason it's no longer relevant?)

Obviously I'm somewhat less concerned about the first bullet, but the second one seems worrying. I'm not super familiar with all this, so maybe I'm overlooking something...